src/core/Entry.cpp

```diff
@@ -670,6 +672,102 @@ void Entry::updateModifiedSinceBegin()
     m_modifiedSinceBegin = true;
 }
 
+QString Entry::resolveMultiplePlaceholdersRecursive(const QString &str, int maxDepth) const
+{
+    if (maxDepth <= 0) {
+        qWarning("Maximum depth of replacement has been reached. Entry uuid: %s", qPrintable(uuid().toHex()));
+        return str;
+    }
+
+    QString result = str;
+    QRegExp placeholderRegEx("(\\{.*\\})", Qt::CaseInsensitive, QRegExp::RegExp2);
+    placeholderRegEx.setMinimal(true);
+    int pos = 0;
+    while ((pos = placeholderRegEx.indexIn(str, pos)) != -1) {
+        const QString found = placeholderRegEx.cap(1);
+        result.replace(found, resolvePlaceholderRecursive(found, maxDepth - 1));
+        pos += placeholderRegEx.matchedLength();
+    }
+
+    if (result != str) {
+        result = resolveMultiplePlaceholdersRecursive(result, maxDepth - 1);
+    }
+
+    return result;
+}
+
+QString Entry::resolvePlaceholderRecursive(const QString &placeholder, int maxDepth) const
+{
+    const PlaceholderType placeholderType = this->placeholderType(placeholder);
```

Comment:
this pointer is not needed.

Reply:
This is useful for debugging, have the value which is returned by the function in a separate variable. I renamed the local variable.

Reply:
I meant the explicit `this` pointer is not needed. Just call `placeholderType()` without `this`.

Reply:
This code didn't work, variable name overrides the function name  
```cpp
const PlaceholderType placeholderType = placeholderType(placeholder); 
```

Reply:
Rename the variable then.